lib/ansible/modules/cloud/amazon/ec2_vpc_net_facts.py

```diff
@@ -57,77 +60,183 @@
+        # We have to make two separate calls per VPC to get these attributes.
+        try:
+            dns_support = connection.describe_vpc_attribute(VpcId=vpc['VpcId'], Attribute='enableDnsSupport')
+            dns_hostnames = connection.describe_vpc_attribute(VpcId=vpc['VpcId'], Attribute='enableDnsHostnames')
+        except (ClientError, NoCredentialsError) as e:
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
 
-    for vpc in all_vpcs:
-        vpc_dict_array.append(get_vpc_info(vpc))
+        # loop through the ClassicLink Enabled results and add the value for the correct VPC
+        for item in cl_enabled['Vpcs']:
+            if vpc['VpcId'] == item['VpcId']:
+                vpc['ClassicLinkEnabled'] = item['ClassicLinkEnabled']
 
-    module.exit_json(vpcs=vpc_dict_array)
+        # loop through the ClassicLink DNS support results and add the value for the correct VPC
+        for item in cl_dns_support['Vpcs']:
+            if vpc['VpcId'] == item['VpcId']:
+                vpc['ClassicLinkDnsSupported'] = item['ClassicLinkDnsSupported']
 
+        # add the two DNS attributes
+        vpc['EnableDnsSupport'] = dns_support['EnableDnsSupport'].get('Value')
+        vpc['EnableDnsHostnames'] = dns_hostnames['EnableDnsHostnames'].get('Value')
+        # for backwards compatibility
+        vpc['id'] = vpc['VpcId']
+        vpc_info.append(camel_dict_to_snake_dict(vpc))
+        # convert tag list to ansible dict
+        vpc_info[-1]['tags'] = boto3_tag_list_to_ansible_dict(vpc['Tags'])
 
-def main():
-    argument_spec = ec2_argument_spec()
-    argument_spec.update(
-        dict(
-            filters = dict(default=None, type='dict')
-        )
-    )
+    module.exit_json(vpcs=vpc_info)
 
-    module = AnsibleModule(argument_spec=argument_spec)
 
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        vpc_ids=dict(type='list', default=[]),
+        filters=dict(type='dict', default={})
+    ))
 
-    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
 
-    if region:
-        try:
-            connection = connect_to_aws(boto.vpc, region, **aws_connect_params)
-        except (boto.exception.NoAuthHandlerFound, StandardError) as e:
-            module.fail_json(msg=str(e))
-    else:
-        module.fail_json(msg="region must be specified")
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required for this module')
 
-    list_ec2_vpcs(connection, module)
+    try:
+        region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+        connection = boto3_conn(module, conn_type='client', resource='ec2', **aws_connect_params)
+    except (ClientError, NoCredentialsError) as e:
```

Comment:
Can that raise a ClientError? I don't think boto3 verifies the connection like boto does when it's first created.